File: collaborators/BryanYunis/solutions/graph-transitiveClosure.js
function closure(graph) {
    const reachable = new Array(graph.length).fill(new Array(graph.length).fill(0))

    for (let i = 0; i < graph.length; i++) {
        for (let j = 0; j < graph[i].length; j++) {
            reachable[i][graph[i][j]] = 1
        }
    }
    console.log(reachable)
    for (let k = 0; k < graph.length; k++) {
        for (let i = 0; i < graph.length; i++) {
            for (let j = 0; j < graph.length; j++) {
                if (reachable[i][k] && reachable[k][j]) {
                    reachable[i][j] = 1
                }
            }
        }
    }
    // return reachable
}

closure([
    [0, 1, 3],
    [1, 2],
    [2],
    [3]
])

File: Contributors/BryanYunis/solutions/graphTransitiveClosure.py
def closure(graph):
    n = len(graph)
    reachable = [[0 for _ in range(n)] for _ in range(n)]

    for i, v in enumerate(graph):
        for neighbor in v:
            reachable[i][neighbor] = 1

    for k in range(n):
        for i in range(n):
            for j in range(n):
                reachable[i][j] |= (reachable[i][k] and reachable[k][j])

    return reachable